tests/functional/algos/test_algos.py

```diff
@@ -247,3 +247,64 @@ def test_with_evc(algorithm):
         param = best_trial._params[1]
         assert param.name == "x"
         assert param.type == "real"
+
+
+@pytest.mark.parametrize(
+    "algorithm", algorithm_configs.values(), ids=list(algorithm_configs.keys())
+)
+def test_parallel_workers(algorithm, tmp_path):
+    """Test parallel execution with joblib"""
+    pickle_host = str(tmp_path / "db.pkl")
+
+    name = "{}_exp".format(algorithm)
+
+    storage = {"type": "legacy", "database": {"type": "PickledDB", "host": pickle_host}}
+    base_exp = create_experiment(
+        name=name,
+        space=space_with_fidelity,
+        algorithms=algorithm_configs["random"],
+        storage=storage,
+    )
+    base_exp.workon(rosenbrock, max_trials=10)
+
+    exp = create_experiment(
+        name=name,
+        space=space_with_fidelity,
+        algorithms=algorithm,
+        branching={"branch_from": name},
+        storage=storage,
+    )
+
+    assert exp.version == 2
+
+    exp.workon(rosenbrock, max_trials=30, n_workers=3)
+
+    assert exp.configuration["algorithms"] == algorithm
+
+    trials = exp.fetch_trials(with_evc_tree=False)
+    assert len(trials) >= 20
+
+    trials_with_evc = exp.fetch_trials(with_evc_tree=True)
+    assert len(trials_with_evc) >= 30
```

Comment:
```suggestion
    assert 30 <=len(trials_with_evc) <= 30 + N_WORKERS
```

Reply:
Change to verify the `completed_trials` as `trials_with_evc` including all the state trials.